src/GeneratorProcessPluginManager.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Drupal\migrate_generator;
+
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Plugin\DefaultPluginManager;
+use Drupal\Component\Plugin\FallbackPluginManagerInterface;
+use Drupal\migrate_generator\Plugin\GeneratorProcessPluginInterface;
+use Drupal\migrate_generator\Annotation\GeneratorProcessPlugin;
+
+/**
+ * Plugin manager for generator process plugins.
+ *
+ * @see \Drupal\migrate_generator\Annotation\GeneratorProcessPlugin
+ * @see \Drupal\migrate_generator\Plugin\GeneratorProcessPluginBase
+ * @see \Drupal\migrate_generator\Plugin\GeneratorProcessPluginInterface
+ * @see plugin_api
+ */
+class GeneratorProcessPluginManager extends DefaultPluginManager implements FallbackPluginManagerInterface {
+
+  /**
+   * Constructs a new GeneratorProcessPluginManager.
+   *
+   * @param \Traversable $namespaces
+   *   An object that implements \Traversable which contains the root paths
+   *   keyed by the corresponding namespace to look for plugin implementations.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache_backend
+   *   Cache backend instance to use.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler to invoke the alter hook with.
+   */
+  public function __construct(\Traversable $namespaces, CacheBackendInterface $cache_backend, ModuleHandlerInterface $module_handler) {
+    parent::__construct('Plugin/migrate_generator/process', $namespaces, $module_handler, GeneratorProcessPluginInterface::class, GeneratorProcessPlugin::class);
+
+    $this->alterInfo('migrate_generator_process_info');
+    $this->setCacheBackend($cache_backend, 'migrate_generator_plugins_process');
+  }
+
+  /**
+   * Check if plugin exists.
+   *
+   * @param string $plugin_id
+   *   A plugin id.
+   *
+   * @return mixed
+   *   A plugin definition, or NULL if the plugin ID is invalid and
+   *   $exception_on_invalid is FALSE.
```

Comment:
We don't have `$exception_on_invalid` on this function, I believe it is a copy/paste error from https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Plugin%21Discovery%21DiscoveryCachedTrait.php/function/DiscoveryCachedTrait%3A%3AgetDefinition/8.2.x so please fix it

Reply:
updated